airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/composite_error_handler.py

```diff
@@ -45,19 +43,30 @@ def __post_init__(self, parameters: Mapping[str, Any]) -> None:
             raise ValueError("CompositeErrorHandler expects at least 1 underlying error handler")
 
     @property
-    def max_retries(self) -> Union[int, None]:
-        return self.error_handlers[0].max_retries
+    def max_retries(self) -> Optional[int]:
+        return self.error_handlers[0].max_retries  # type: ignore # property not defined in ErrorHandler
 
     @property
     def max_time(self) -> Union[int, None]:
-        return max([error_handler.max_time or 0 for error_handler in self.error_handlers])
-
-    def interpret_response(self, response: requests.Response) -> ResponseStatus:
-        should_retry = ResponseStatus(ResponseAction.FAIL)
-        for retrier in self.error_handlers:
-            should_retry = retrier.interpret_response(response)
-            if should_retry.action == ResponseAction.SUCCESS:
-                return response_status.SUCCESS
-            if should_retry == response_status.IGNORE or should_retry.action == ResponseAction.RETRY:
-                return should_retry
-        return should_retry
+        return max([error_handler.max_time or 0 for error_handler in self.error_handlers])  # type: ignore # property not defined in ErrorHandler
+
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]]) -> ErrorResolution:
+        matched_error_resolution = None
+        for error_handler in self.error_handlers:
+            matched_error_resolution = error_handler.interpret_response(response_or_exception)
+
+            if not isinstance(matched_error_resolution, ErrorResolution):
+                continue
+
+            if matched_error_resolution.response_action == ResponseAction.SUCCESS:
+                return matched_error_resolution
+
+            if (
+                matched_error_resolution.response_action == ResponseAction.RETRY
+                or matched_error_resolution.response_action == ResponseAction.IGNORE
+            ):
+                return matched_error_resolution
+        if matched_error_resolution:
+            return matched_error_resolution
```

Comment:
```suggestion
            if not isinstance(matched_error_resolution, ErrorResolution):
                continue
            else:
                return matched_error_resolution
```

Reply:
The thought behind the current indentation levels/response hierarchies is:

1. Iterate over each error handler to determine:
    a. If response is SUCCESS, return success resolution
    b. If response is RETRY or IGNORE, return resolution
    c. If response is FAIL, iterate to next error handler to see if there is a "better" specified action for the given response
2. After iterating over all error handlers, if there is a matched error resolution, (it would presumably be a FAIL response action, otherwise it would have already been returned), then return the resolution
3. Otherwise, return a default error resolution (RETRY w/ system_error)

The thought here is we don't want to prematurely fail if there is an error handler with a response filter that has a "better" outcome. IMO this makes a lot of assumptions about the connector dev's intentions, but while the CompositeErrorHandler still exists, I'm trying to maintain its behavior as much as possible.

Reply:
Then should it be like this?
```
    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]]) -> ErrorResolution:
        best_resolution = None

        for error_handler in self.error_handlers:
            error_resolution = error_handler.interpret_response(response_or_exception)

            # Check if error_resolution is valid and prioritize based on action
            if isinstance(error_resolution, ErrorResolution):
                if error_resolution.response_action == ResponseAction.SUCCESS:
                    return error_resolution  # Immediate return on success
                elif error_resolution.response_action in (ResponseAction.RETRY, ResponseAction.IGNORE):
                    best_resolution = error_resolution  # Store RETRY or IGNORE as potential best outcomes
                elif error_resolution.response_action == ResponseAction.FAIL and not best_resolution:
                    best_resolution = error_resolution  # Store FAIL only if no better alternative has been found

        # Return the best resolution found, default to RETRY if none provided
        return best_resolution if best_resolution else ErrorResolution(ResponseAction.RETRY)
        ```
Why do we not have the same implementation for DefaultErrorHandler?